main/src/library/File.flix

```diff
@@ -364,4 +364,152 @@ namespace File {
             }
         }
 
+    ///
+    /// Writes `data` to the given file `f`.
+    ///
+    /// Creates file `f` if it does not exist. Overwrites it if it exists.
+    ///
+    /// Returns `true` if the file `f` was created, and `false` if `f` was overwritten.
+    ///
+    pub def write(f: String, data: t): Result[Bool, String] & Impure with ToString[t] =
+        try {
+            import new java.io.File(String) as newFile;
+            import new java.io.PrintWriter(##java.io.File) as newPrintWriter;
+            import java.io.PrintWriter.println(String);
+            import java.io.PrintWriter.checkError();
+
+            let alreadyExists = File.exists(f);
+
+            let javaFile = newFile(f);
+            let w = newPrintWriter(javaFile);
+
+            println(w, "${data}");
+
+            if (checkError(w)) {
+                Err("An error happened when writing to the file: "+f)
+            }
+            else {
+                match alreadyExists {
+                    case Ok(exists) => Ok(not exists)
+                    case Err(mes)   => Err(mes)
+                }
```

Comment:
This looks like a good place to use either `Result.flatMap` or the `let*` syntax. Are you familiar with those?

Reply:
Just looked at `Result.flatMap`. That would be very useful here